.github/workflows/main.yml

```diff
@@ -15,38 +15,45 @@ jobs:
   # This workflow contains a single job called "build"
   build:
     # This allows us to find the workflow in checks
-    name: ci-test-build
+    name: ci-test
     
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # python-version: ['3.7', '3.8', '3.9']
+        python-version: ['3.7', '3.8']
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
     
     # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
     
     # Ensure we are running Python v3.x
-    - name: Set up Python
-      uses: actions/setup-python@v1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
       with:
-        # Issues with bdist_wheel on v3.9
-        #python-version: '3.x'
-        python-version: '3.7'
+        python-version: ${{ matrix.python-version }}
     
     # Install ASE so we can run our tests 
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install --upgrade ase==3.22.1
-        python -m pip install --upgrade catlearn
+        python -m pip install --upgrade ase
+        #Upgrading Catlearn to the most recent version 
+        python -m pip install https://github.com/SUNCAT-Center/CatLearn/archive/refs/heads/master.zip
         python -m pip install --upgrade ase-gpatom
         python -m pip install --upgrade pytest
         python -m pip install --upgrade pytest-cov
-        python -m pip install torch torchvision torchaudio
-        python -m pip install --upgrade mace-torch
-        python -m pip install pymatgen
+        # Removing as presumably MACE will automatically install dependencies
+        # python -m pip install torch torchvision torchaudio
+        # Restricted MACE as > 0.3.4 fails on Python 3.7
+        python -m pip install mace-torch==0.3.4 
+        python -m pip install --upgrade pymatgen
         
+        # Debug
+        python -m pip freeze
```

Comment:
Do we want to keep the debug?

Reply:
I think no harm to keep it - means we can see versions if any questions arise, and low cost.